# Dataset sample Waternet 

In [2]:
# Import the dataset sample file into adataframe and name the dataframe 'waternet'

import pandas as pd
import numpy as np
import datetime as dt

waternet = pd.read_csv ("CSV.csv", sep=";")

In [3]:
# This is all okay, but...
pip install missingno

Note: you may need to restart the kernel to use updated packages.


In [ ]:
# As soon as I import missingno, the kernel dies (see top left corner of screen, saying Trusted now)
# There are no search hits on Google on import missingno and dead kernel
import missingno

In [3]:
# Checking whether the data is loaded succesfully in the dataframe
print(waternet[:5])

           Tijd 5018 putniveau gemaal Overstort 623 Overstort 624 BBB riool  \
0           NaN                  mNAP          mNAP          mNAP      mNAP   
1  1-01-16 0:00                 -2,63        -1,749        -1,539    -2,526   
2  1-01-16 0:05                 -2,64        -1,749        -1,539    -2,526   
3  1-01-16 0:10                 -2,66        -1,749        -1,539    -2,526   
4  1-01-16 0:15                 -2,61        -1,749        -1,539    -2,526   

  BBB bassin begin BBB bassin eind BBB-buffer  \
0             mNAP            mNAP       mNAP   
1           -4,166          -3,302     -2,292   
2           -4,166          -3,302     -2,292   
3           -4,166          -3,302     -2,292   
4           -4,166          -3,302     -2,292   

  Neerslag Papaverweg (grondstation)  \
0                      mm/ 5 minuten   
1                              0,000   
2                              0,000   
3                              0,000   
4                             

### We see that the data matches the data in Excel! Joepie


#### In this dataset:

The columns display information about:
- Tijd = the date and time of the measurements (each row has an interval of 5 min)
- 5018 putniveau gemaal (mNAP)= water pomping station 5018 in the middle of the peninsula
- Overstort 623 (mNAP) = overflow basin in the north-east of the peninsula
- Overstort 624 (mNAP) = overflow basin in the far west of the peninsula
- BBB riool (mNAP) = 
- BBB basin begin (mNAP) =
- BBB basin eind (mNAP) = 
- BBB-buffer (mNAP) = 
- Neerslag Papaverweg (grondstation) (mm/5 minuten) = 
- Neerslag Buikslotermeerplein (radar Hydronet) (mm/5 minuten) =

In [4]:
print(len(waternet))

526177


The dataset has a total of 526177 rows, which is correct, because the number of rows in Excel is 526178 and the first row there begins with index number 1, and in here it is index number 0

In [5]:
# Lets clean up the names of the header a little bit, so we can work with it better

waternet.rename (columns = {"Tijd": "Time"}, inplace = True)
waternet.rename (columns = {"5018 putniveau gemaal": "water_pomping_station"}, inplace = True)
waternet.rename (columns = {"Overstort 623": "basin_northeast"}, inplace = True)
waternet.rename (columns = {"Overstort 624": "basin_west"}, inplace = True)
waternet.rename (columns = {"BBB riool": "sewer"}, inplace = True)
waternet.rename (columns = {"BBB bassin begin": "last_basin_start"}, inplace = True)
waternet.rename (columns = {"BBB bassin eind": "last_basin_end"}, inplace = True)
waternet.rename (columns = {"BBB-buffer": "last_basin_buffer"}, inplace = True)
waternet.rename (columns = {"Neerslag Papaverweg (grondstation)": "rainfall_groundstation"}, inplace = True)
waternet.rename (columns = {"Neerslag Buikslotermeerplein (radar Hydronet)": "rainfall_radar"}, inplace = True)

In [6]:
print(waternet.head())

           Time water_pomping_station basin_northeast basin_west   sewer  \
0           NaN                  mNAP            mNAP       mNAP    mNAP   
1  1-01-16 0:00                 -2,63          -1,749     -1,539  -2,526   
2  1-01-16 0:05                 -2,64          -1,749     -1,539  -2,526   
3  1-01-16 0:10                 -2,66          -1,749     -1,539  -2,526   
4  1-01-16 0:15                 -2,61          -1,749     -1,539  -2,526   

  last_basin_start last_basin_end last_basin_buffer rainfall_groundstation  \
0             mNAP           mNAP              mNAP          mm/ 5 minuten   
1           -4,166         -3,302            -2,292                  0,000   
2           -4,166         -3,302            -2,292                  0,000   
3           -4,166         -3,302            -2,292                  0,000   
4           -4,166         -3,302            -2,292                  0,000   

  rainfall_radar  
0  mm/ 5 minuten  
1          0,000  
2          0,000 

In [7]:
# We need to replace the comma's with dots to make sure we can read the data correctly
waternet["water_pomping_station"] = waternet["water_pomping_station"].str.replace (",", ".")
waternet["basin_northeast"] = waternet["basin_northeast"].str.replace (",", ".")
waternet["basin_west"] = waternet["basin_west"].str.replace (",", ".")
waternet["sewer"] = waternet["sewer"].str.replace (",", ".")
waternet["last_basin_start"] = waternet["last_basin_start"].str.replace (",", ".")
waternet["last_basin_end"] = waternet["last_basin_end"].str.replace (",", ".")
waternet["last_basin_buffer"] = waternet["last_basin_buffer"].str.replace (",", ".")
waternet["rainfall_groundstation"] = waternet["rainfall_groundstation"].str.replace (",", ".")
waternet["rainfall_radar"] = waternet["rainfall_radar"].str.replace (",", ".")

In [8]:
print(waternet.describe)

<bound method NDFrame.describe of                   Time water_pomping_station basin_northeast basin_west  \
0                  NaN                  mNAP            mNAP       mNAP   
1         1-01-16 0:00                 -2.63          -1.749     -1.539   
2         1-01-16 0:05                 -2.64          -1.749     -1.539   
3         1-01-16 0:10                 -2.66          -1.749     -1.539   
4         1-01-16 0:15                 -2.61          -1.749     -1.539   
...                ...                   ...             ...        ...   
526172  31-12-20 23:35                 -2.48          -1.749     -1.539   
526173  31-12-20 23:40                 -2.39          -1.749     -1.539   
526174  31-12-20 23:45                 -2.48          -1.749     -1.539   
526175  31-12-20 23:50                 -2.40          -1.749     -1.539   
526176  31-12-20 23:55                 -2.49          -1.749     -1.539   

         sewer last_basin_start last_basin_end last_basin_buffer 

In [9]:
# Explore the number of unique values in the water_pomping_station column
waternet["water_pomping_station"].unique().shape

(183,)

In [10]:
# Explore the count, mininimum, maximum, median, mean, etc. in the water_pomping_station column

waternet["water_pomping_station"].describe

<bound method NDFrame.describe of 0          mNAP
1         -2.63
2         -2.64
3         -2.66
4         -2.61
          ...  
526172    -2.48
526173    -2.39
526174    -2.48
526175    -2.40
526176    -2.49
Name: water_pomping_station, Length: 526177, dtype: object>

183 unique values seems pretty good, indicating enough variation at the source of our sewer system

In [11]:
# Now lets explore the number of unique values in the basin_norhteast column

waternet["basin_northeast"].unique().shape

(381,)

Also looks pretty good. Like someting is happening there at least.

In [12]:
# Lets do the same for the number of unique values in the basin_west column

waternet["basin_west"].unique().shape

(307,)

Same applies to this basin as well

In [13]:
# Lets do the same for the number of unique values in the sewer column

waternet["sewer"].unique().shape

(1174,)

In [14]:
# Lets do the same for the number of unique values in the last_basin_start column

waternet["last_basin_start"].unique().shape

(2029,)

In [15]:
# Lets do the same for the number of unique values in the last_basin_end column

waternet["last_basin_buffer"].unique().shape

(985,)

In [16]:
# Lets do the same for the number of unique values in the rainfall_groundstation column

waternet["rainfall_groundstation"].unique().shape

(62,)

In [17]:
# Lets do the same for the number of unique values in the rainfall_radar column

waternet["rainfall_radar"].unique().shape

(198,)

In [18]:
missingno.matrix(waternet)

NameError: name 'missingno' is not defined

In [19]:
waternet.isnull().sum(axis=1)

0         1
1         0
2         0
3         0
4         0
         ..
526172    0
526173    0
526174    0
526175    0
526176    0
Length: 526177, dtype: int64

In [1]:
# You see here that the number of missing values in the columns is none